dht/token.hh

```diff
@@ -84,6 +84,15 @@ public:
         std::copy_n(reinterpret_cast<int8_t*>(&t), 8, b.begin());
         return b;
     }
+
+    /**
+     * Calculate a token representing the approximate "middle" of the given
+     * range.
+     *
+     * @return The approximate midpoint between left and right.
+     */
+    static token midpoint(const token& left, const token& right);
```

Comment:
Why make it a static method instead of just a function?

Reply:
It is more discoverable. I like free functions myself but agree that discoverability is better as a member.